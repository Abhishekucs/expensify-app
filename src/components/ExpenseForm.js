import React from 'react'
import moment from 'moment'
import 'react-dates/initialize'
import { SingleDatePicker } from 'react-dates'

export default class ExpenseForm extends React.Component {
    constructor(props){
        super(props)

        this.state = {
            description: props.expense ? props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense ? props.expense.amount.toString() : '',
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calenderFocused:false,
            error: ''
        }
    }
    onAmountChange = (e) => {
        const amount = e.target.value
        if(!amount || amount.match(/^\d*(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount }))
        }
    }
    onDateChange = (createdAt) => {
        if(createdAt) {
            this.setState(() => ({ createdAt }))
        }
    }
    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calenderFocused: focused }))
    }
    onSubmit = (e) => {
        e.preventDefault()

        if(!this.state.description || !this.state.amount) {
            this.setState(() => ({ error: 'Please provide description and amount' }))
        }else {
            this.setState(() => ({ error: '' }))
            this.props.onSubmit(
                {
                    description: this.state.description,
                    amount: parseFloat(this.state.amount),
                    note: this.state.note,
                    createdAt: this.state.createdAt.valueOf()
                }
            )
        }
    }
    render() {
        return (
                <form className="form" onSubmit={this.onSubmit}>
                {this.state.error && <p className="form__error">{this.state.error}</p>}
                    <input 
                        type="text"
                        className="text-input"
                        placeholder="Description"
                        autoFocus={true}
                        value = {this.state.description}
                        onChange = {(e) => {
                            const description = e.target.value
                            this.setState(() => ({ description }))
                        }}
                    />
                    <input 
                        type="text"
                        className="text-input"
                        placeholder="Amount"
                        value = {this.state.amount}
                        onChange = {this.onAmountChange}
                    />
                    <SingleDatePicker 
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.calenderFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                    />
                    <textarea
                        placeholder="Add note(optionally)"
                        className="textarea"
                        value = {this.state.note}
                        onChange = {(e) => {
                            const note = e.target.value
                            this.setState(() => ({ note }))
                        }}
                    >
                    </textarea>
                    <div>
                        <button className="button">Save Expense</button>
                    </div>
                </form>
        )
    }
}