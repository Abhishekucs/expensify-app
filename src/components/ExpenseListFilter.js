import React from 'react'
import { connect } from 'react-redux'
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters'
import 'react-dates/initialize'
import { DateRangePicker } from 'react-dates'

export class ExpenseListFilters extends React.Component {
    state = {
        focusedInput: null
    }
    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }
    onFocusChange = (focusedInput) => {
        this.setState(() => ({ focusedInput }))
    }
    onTextChange = (e) => {
        const value = e.target.value
        this.props.setTextFilter({ text: value })
    }
    onSortChange = (e) => {
        if(e.target.value === 'amount'){
            this.props.sortByAmount()
        } else {
            this.props.sortByDate()
        }
    }
    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input className="text-input" type="text" placeholder="Search Expenses" value={this.props.filters.text} onChange={this.onTextChange} />
                    </div>
                    <div className="input-group__item">
                        <select className="select" value={this.props.filters.sortBy} onChange={this.onSortChange}>
                            <option value="date">Date</option>
                            <option value="amount">Amount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker 
                            startDate = {this.props.filters.startDate}
                            startDateId = "myStartDateId"
                            endDate = {this.props.filters.endDate}
                            endDateId = "myEndDateId"
                            onDatesChange = {this.onDatesChange}
                            focusedInput = {this.state.focusedInput}
                            onFocusChange = {this.onFocusChange}
                            numberOfMonths = {1}
                            isOutsideRange = {() => false}
                            showClearDates = {true}
                        />
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        setTextFilter: ({text}) => dispatch(setTextFilter({text})),
        setStartDate: (startDate) => dispatch(setStartDate(startDate)),
        setEndDate: (endDate) => dispatch(setEndDate(endDate)),
        sortByAmount: () => dispatch(sortByAmount()),
        sortByDate: () => dispatch(sortByDate())
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)