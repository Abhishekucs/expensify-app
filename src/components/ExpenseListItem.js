import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'

const ExpenseListItem = ({ id, description, createdAt, amount }) => (
    <Link className="list-item" to={`/edit/${id}`}>
        <div>
            <h3 className="list-item__title">{description}</h3>
            <span className="list-item__sub-title">{moment(createdAt).format('MMM Do, YYYY')}</span>
        </div>
        <h3 className="list-item__data"><span>Rs</span> {new Intl.NumberFormat('en-IN').format(amount)}</h3>
    </Link>
)

export default ExpenseListItem