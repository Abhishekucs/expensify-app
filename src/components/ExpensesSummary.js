import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import getExpensesTotal from '../selectors/expenses-total'
import visibleExpenseData from '../selectors/expense'

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{ new Intl.NumberFormat('en-IN').format(expensesTotal) }</span></h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const expenseData = visibleExpenseData(state.expenses, state.filters)
    return {
        expenseCount: expenseData.length,
        expensesTotal: getExpensesTotal(expenseData)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)