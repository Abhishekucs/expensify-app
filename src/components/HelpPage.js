import React from 'react'

const HelpPage = () => (
    <div>
        <p>This is from Help Page</p>
    </div>
)

export default HelpPage