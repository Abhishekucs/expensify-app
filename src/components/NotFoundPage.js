import React from 'react'
import { Link } from 'react-router-dom'

const NotFoundPage = () => (
    <div>
        <p>404!</p>
        <Link to="/">Go Home</Link>
    </div>
)

export default NotFoundPage