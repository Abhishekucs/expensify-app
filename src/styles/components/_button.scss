.button {
    background: $blue;
    border: none;
    padding: $s-size;
    color: white;
    display: inline-block;
    font-weight: 300;
    font-size: $font-size-large;
    text-decoration: none;
    line-height: 1;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888;
}