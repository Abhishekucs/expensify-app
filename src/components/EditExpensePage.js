import React from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { startEditExpenses } from '../actions/expenses'
import { startRemoveExpenses } from '../actions/expenses'

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpenses(this.props.expense.id ,expense)
        this.props.history.push('/')
    }
    onClick = (e) => {
        this.props.startRemoveExpenses( {id: this.props.expense.id} )
        this.props.history.push('/')
    }
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm 
                        expense = {this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button className="button button--secondary" onClick ={this.onClick}>Remove Expense</button>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => {
            return expense.id === props.match.params.id
        })
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        startEditExpenses: (id, expense) => dispatch(startEditExpenses(id, expense)),
        startRemoveExpenses: (data) => dispatch(startRemoveExpenses(data))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)