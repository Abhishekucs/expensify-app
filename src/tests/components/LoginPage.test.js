import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { LoginPage } from '../../components/LoginPage'

test('should render Login Page', () => {
    const wrapper = shallow(<LoginPage />)
    expect(toJSON(wrapper)).toMatchSnapshot()
})

test('should startLogin on button click', () => {
    const startLogin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLogin}/>)
    wrapper.find('button').simulate('click')
    expect(startLogin).toHaveBeenCalled();
    expect(toJSON(wrapper)).toMatchSnapshot();
})