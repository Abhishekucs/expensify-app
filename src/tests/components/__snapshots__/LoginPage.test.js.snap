// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login Page 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify-App
    </h1>
    <p>
      Its time to get your expenses under control.
    </p>
    <button
      className="button"
    >
      Login with Google
    </button>
  </div>
</div>
`;

exports[`should startLogin on button click 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify-App
    </h1>
    <p>
      Its time to get your expenses under control.
    </p>
    <button
      className="button"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    >
      Login with Google
    </button>
  </div>
</div>
`;
